src/lib/spinel/radio_spinel.cpp

```diff
@@ -1775,6 +1777,57 @@ void RadioSpinel::GetDiagOutputCallback(otPlatDiagOutputCallback &aCallback, voi
     aContext  = mOutputContext;
 }
 
+otError RadioSpinel::PlatDiagProcess(uint8_t aArgsLength, char *aArgs[])
+{
+    otError error                                            = OT_ERROR_NONE;
+    char    cmd[OPENTHREAD_CONFIG_DIAG_CMD_LINE_BUFFER_SIZE] = {'\0'};
+    char   *cur                                              = cmd;
+    char   *end                                              = cmd + sizeof(cmd);
+
+    // RadioSpinel-specific diag sub-commands
+    if (strcmp(aArgs[0], "radiospinel") == 0)
```

Comment:
How about add a new function to process the `radiospinel` command?

Reply:
SGTM, added `RadioSpinelDiagProcess()` . I've moved check for the first argument to `radio.cpp` as `rcpcaps` are checked there as well, so it wouldn't be scattered over multiple files/functions.